compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeResolver.java

```diff
@@ -1457,14 +1457,20 @@ private void fillEffectiveType(BIntersectionType intersectionType,
         }
 
         while (iterator.hasNext()) {
+            bLangEffectiveImpliedType = Types.getImpliedType(effectiveType);
```

Comment:
Since we are adding this code block here, we can remove L1451 - L1457

Reply:
Address in [commit](https://github.com/ballerina-platform/ballerina-lang/pull/42025/commits/e2446c6b23061200c3f21ee681f88f68373b25de).